python/GafferUI/SpreadsheetUI.py

```diff
@@ -1212,6 +1315,276 @@ def __fixedWidth( cls, plugValueWidget ) :
 
 		return None
 
+# _SectionChooser
+# ===============
+
+class _SectionChooser( GafferUI.Widget ) :
+
+	def __init__( self, rowsPlug, **kw ) :
+
+		tabBar = QtWidgets.QTabBar()
+		GafferUI.Widget.__init__( self, tabBar, **kw )
+
+		self.__rowsPlug = rowsPlug
+		self.__currentSectionChangedSignal = Gaffer.Signal1()
+
+		tabBar.setDrawBase( False )
+		tabBar.setMovable( True )
+		tabBar.setExpanding( False )
+		tabBar.setUsesScrollButtons( False )
+
+		tabBar.setContextMenuPolicy( QtCore.Qt.CustomContextMenu )
+		tabBar.customContextMenuRequested.connect( Gaffer.WeakMethod( self.__contextMenuRequested ) )
+
+		tabBar.currentChanged.connect( Gaffer.WeakMethod( self.__currentChanged ) )
+		self.__ignoreCurrentChanged = False
+		tabBar.tabMoved.connect( Gaffer.WeakMethod( self.__tabMoved ) )
+		self.__plugMetadataChangedConnection = Gaffer.Metadata.plugValueChangedSignal().connect( Gaffer.WeakMethod( self.__plugMetadataChanged ) )
+		self.__rowsPlug["default"]["cells"].childAddedSignal().connect( Gaffer.WeakMethod( self.__columnAdded ), scoped = False )
+		self.__rowsPlug["default"]["cells"].childRemovedSignal().connect( Gaffer.WeakMethod( self.__columnRemoved ), scoped = False )
+
+		self.__updateTabs()
+
+	def currentSection( self ) :
+
+		if not self._qtWidget().count() :
+			return None
+
+		return self._qtWidget().tabText( self._qtWidget().currentIndex() )
+
+	def currentSectionChangedSignal( self ) :
+
+		return self.__currentSectionChangedSignal
+
+	@classmethod
+	def sectionNames( cls, rowsPlug ) :
+
+		names = set()
+		for cellPlug in rowsPlug["default"]["cells"] :
+			sectionName = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" ) or ""
```

Comment:
Given the current behaviour (in that we only hide the tab bar with no authored sections, rather than just one, and that you can rename 'other' to make it permanent), would it make sense to just use `"Other"` as the default value for the metadata query, rather than the custom logic below? And hide the bar when there is only one section?  Otherwise there is no way to go back to having no sections once you've added one.

Reply:
Tidied this all up in d35774d, which uses `setSection()/getSection()` instead of direct metadata access.